src/app/services/announcement/announcement.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@angular/core';
+import { ReplaySubject } from 'rxjs';
+import { Announcement, AnnouncementType } from './announcement';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AnnouncementService {
+  readonly newAnnouncement$: ReplaySubject<Announcement>;
+
+  constructor() {
+    this.newAnnouncement$ = new ReplaySubject<Announcement>(1);
```

Comment:
What is the '1' doing on this line?

Reply:
ReplaySubject buffers the n previous results. We want to just buffer 1 to handle race conditions where the announcement is made prior to the DOM being set up.